pages/settingspage.py

```diff
@@ -106,32 +123,29 @@ def set_data(self):
             self.buy_one_checkbox.setChecked(True)
         if settings['dont_buy']:
             self.dont_buy_checkbox.setChecked(True)
+        self.random_delay_start.setText(settings["random_delay_start"])
+        self.random_delay_stop.setText(settings["random_delay_stop"])
         self.update_settings(settings)
 
     def save_settings(self):
-        settings = {"webhook":self.webhook_edit.text(),
-                    "webhookonbrowser":self.browser_checkbox.isChecked(),
-                    "webhookonorder":self.order_checkbox.isChecked(),
-                    "webhookonfailed":self.paymentfailed_checkbox.isChecked(),
-                    "browseronfailed":self.onfailed_checkbox.isChecked(),
-                    "onlybuyone":self.buy_one_checkbox.isChecked(),
-                    "dont_buy":self.dont_buy_checkbox.isChecked()}
+        settings = {"webhook":            self.webhook_edit.text(),
+                    "webhookonbrowser":   self.browser_checkbox.isChecked(),
+                    "webhookonorder":     self.order_checkbox.isChecked(),
+                    "webhookonfailed":    self.paymentfailed_checkbox.isChecked(),
+                    "browseronfailed":    self.onfailed_checkbox.isChecked(),
+                    "onlybuyone":         self.buy_one_checkbox.isChecked(),
+                    "dont_buy":           self.dont_buy_checkbox.isChecked(),
+                    "random_delay_start": self.random_delay_start.text(),
+                    "random_delay_stop":  self.random_delay_stop.text()}
         write_data("./data/settings.json",settings)
         self.update_settings(settings)
         QtWidgets.QMessageBox.information(self, "Phoenix Bot", "Saved Settings")
 
-    def update_settings(self,settings_data):
-        global webhook, webhook_on_browser, webhook_on_order, webhook_on_failed, browser_on_failed, dont_buy
+    def update_settings(self, settings_data):
+        global webhook, webhook_on_browser, webhook_on_order, webhook_on_failed, browser_on_failed, dont_buy, random_delay_start, random_delay_stop
         settings.webhook, settings.webhook_on_browser, settings.webhook_on_order, settings.webhook_on_failed, settings.browser_on_failed, settings.buy_one, settings.dont_buy = settings_data["webhook"], settings_data["webhookonbrowser"], settings_data["webhookonorder"], settings_data["webhookonfailed"], settings_data["browseronfailed"], settings_data['onlybuyone'], settings_data['dont_buy']
 
-
-
-
-
-
-
-
-
-
-
-
+        if settings_data["random_delay_start"] != "":
+            settings.random_delay_start = settings_data["random_delay_start"]
+        if settings_data["random_delay_stop"] != "":
+            settings.random_delay_stop = settings_data["random_delay_stop"]
```

Comment:
As to not potentially break people's existing settings. Can we instead make this:
```suggestion
        if settings_data.get("random_delay_start", "") != "":
            settings.random_delay_start = settings_data["random_delay_start"]
        if settings_data.get("random_delay_stop", "") != "":
            settings.random_delay_stop = settings_data["random_delay_stop"]
```